server/v2/cometbft/commands.go

```diff
@@ -21,22 +21,30 @@ import (
 	"cosmossdk.io/server/v2/cometbft/client/rpc"
 	"cosmossdk.io/server/v2/cometbft/flags"
 	auth "cosmossdk.io/x/auth/client/cli"
+	"github.com/cosmos/cosmos-sdk/client"
 
 	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/query"
 	"github.com/cosmos/cosmos-sdk/version"
 )
 
-func (s *CometBFTServer[T]) rpcClient() (rpc.CometRPC, error) {
+func (s *CometBFTServer[T]) rpcClient(cmd *cobra.Command) (rpc.CometRPC, error) {
 	if s.config.Standalone {
-		client, err := rpchttp.New(s.config.CmtConfig.RPC.ListenAddress)
+		client, err := rpchttp.New(client.GetConfigFromCmd(cmd).RPC.ListenAddress)
 		if err != nil {
 			return nil, err
 		}
 		return client, nil
 	}
 
+	if s.Node == nil || cmd.Flags().Changed(flags.FlagNode) {
+		rpcURI, _ := cmd.Flags().GetString(flags.FlagNode)
```

Comment:
since we can return an error here i would not silent ignore it; even though it's unlikely to happen due to line above